packages/pwa-kit-extension-sdk/src/react/classes/ApplicationExtension.tsx

```diff
@@ -128,9 +131,11 @@ export class ApplicationExtension<
 
             return {
                 ...route,
-                componentName: route.component.displayName
+                componentName: route.component.displayName,
+                ...(route.componentProps && {componentProps: route.componentProps})
```

Comment:
include `componentProps` in the serialized route if it exists in the route

Reply:
that's interesting the return type of `serializeAsyncRoutes` should be `SerializedRouteProps` which contains `componentProps` maybe you don't have the latest code in your local? 
<img width="702" alt="Screenshot 2025-05-05 at 10 06 05 AM" src="https://github.com/user-attachments/assets/7efe4866-adf3-4026-8901-962ef8eb3edc" />